cmdx.py

```diff
@@ -1898,42 +1981,18 @@ def flatten(self, type=None):
 
         """
 
-        members = set()
-
-        def recurse(objset):
-            for member in objset:
-                if member.isA(om.MFn.kSet):
-                    recurse(member)
-                elif type is not None:
-                    if type == member.typeName:
-                        members.add(member)
-                else:
-                    members.add(member)
-
-        recurse(self)
-
-        return list(members)
+        return iter(self.members(type=type, flatten=True))
```

Comment:
This changes the return type from list to iterator, better stick with list here, especially since we aren't gaining anything from it being an iterator.

Reply:
yeah ok! i didn't paid attention to this. It suffered from the legacy translating. And besides, I added a flatten flag directly to .members() method which is alrerady an iterator